core/src/main/java/org/infinispan/util/concurrent/locks/OwnableReentrantLock.java

```diff
@@ -40,6 +42,7 @@ protected final Object currentRequestor() {
    }
 
    private void setCurrentRequestor(Object requestor) {
+      assert requestorOnStack.get() == null;
```

Comment:
I was unaware that we enabled them on tests.  But this still seems like maybe we should be throwing an IllegalStateException if this occurs ?